src/Appwrite/Platform/Workers/Messaging.php

```diff
@@ -104,4 +93,66 @@ public function action(Message $message): void
             throw new Exception('Error sending message: ' . $error->getMessage(), 500);
         }
     }
+
+    protected static function createFromDSN(DSN $dsn): SMSAdapter
+    {
+        $adapter = null;
+
+        switch ($dsn->getHost()) {
+            case 'mock':
+                $adapter = new Mock($dsn->getUser(), $dsn->getPassword());
+                break;
+            case 'msg91':
+                $adapter = new Msg91($dsn->getUser(), $dsn->getPassword());
+                $adapter->setTemplate($dsn->getParam('template', ''));
```

Comment:
Previously, `_APP_SMS_FROM ` was used as the Flow ID and passed to Msg91. Maybe we should fallback to that for backwards compatibility?

Reply:
👍

Reply:
This poses a larger problem:

- `_APP_SMS_FROM` currently has a single value. GEOSMS local adapters that depend on `_APP_SMS_FROM` will use the same value.
- Local adapters will need different values, as we have different numbers set up with different providers.
- This isn't a problem for MSG91, and maybe some others, because it uses the `username` DSN part as the sender ID/from.

We either need to:
- rethink the DSN approach, 
- update messaging to support `->setFrom()` function on the adapter/or individual providers, and set  `?from=` parameter in the DSN
- leave it for now as we just need MSG91 + Twilio which will work

Reply:
You can set the `from` value per **message** but not adapter, can that work without change in the library if we add it to the DSN?

Reply:
Not really without duplicating the geosms adapter logic in the messaging library

Reply:
Let's add `from` as a default null constructor parameter to SMS adapters and use that instead of the message from if it's set. Not ideal as we'll overriding a value from a smaller scope with the one from larger scope which is counter-intuitive.. can't see any way around it though.